framework/auth/views.py

```diff
@@ -543,6 +526,23 @@ def external_login_confirm_email_get(auth, uid, token):
     user.verification_key = generate_verification_key()
     user.save()
 
+    # if user is already logged in
```

Comment:
Curious of how much of the above should indeed be happening before the current logged in status of the user is checked - for example `user.date_last_logged_in = timezone.now()` should probably still happen after the logged in check.